Kubernetes/windows/debug/dnsmonitor/DnsPktMonitor2019.yaml

```diff
@@ -0,0 +1,128 @@
+            }
+
+            $pods = (crictl pods -o json | ConvertFrom-Json).items
+            $podIds = @()
+            $podIPs = @()
+
+            foreach($pod in $pods) {
+              if($pod.metadata.name -like "$podPrefix*") { 
+                $pod.metadata.name
+                $podIds += $pod.id
+                $podInspect = (crictl inspectp $pod.id | ConvertFrom-Json)
+                $podIPs += $podInspect.status.network.ip
+              } 
+            }
+
+            Write-Host "POD IDS : $podIds"
+            Write-Host "POD IPS : $podIPs"
+
+            $containerIds = @()
+
+            foreach($podId in $podIds) {
+              $containers = (crictl ps -o json | ConvertFrom-Json).containers
+              foreach($container in $containers) {
+                if($container.podSandboxId -eq $podId) {
+                  $containerIds += $container.id
+                }
+              }
+            }
+
+            Write-Host "CONTAINER IDS : $containerIds"
+
+            $macAddrs = @()
+
+            foreach($containerId in $containerIds) {
+              $macStr = crictl exec $containerId ipconfig /all | sls "Physical Address"
+              $macSplitArr = $macStr.ToString().Split(": ")
+              $macAddrs += $macStr.ToString().Split(": ")[$macSplitArr.count-1]
+            }
+
+            Write-Host "MAC ADDRESSES : $macAddrs"
+
+            $compIds = ""
+
+            foreach($mac in $macAddrs) {
+              $grepped = pktmon list | Select-String $mac
+              $compId = $grepped.ToString().Split(" ")[3]
+              if($compId -ne "") {
+                if($compIds -eq "") {
+                  $compIds = $compId
+                } else {
+                  $compIds += ","
+                  $compIds += $compId
+                }
+              }
+            }
+
+            Write-Host "COMPARTMENT IDS : $compIds"
+
+            Write-Host "Removing all pktmon filters if anything existing..."
+            pktmon filter remove
+
+            Write-Host "Create DNS Port filter..."
+            pktmon filter add DNSFilter -p 53
+
+            Write-Host "Create a directory for pktmon logs..."
+            remove-item -Recurse -Force $pktmonLogs -ErrorAction Ignore
+            mkdir $pktmonLogs
+            Set-Location $pktmonLogs
+
+            Write-Host "Start pktmon. Command : [pktmon start -c --comp $compIds -m multi-file] ..."
+            pktmon start -c --comp $compIds -m multi-file
+
+            Write-Host "Logs will be available in $pktmonLogs"
+
+            While($true) {
+              Start-Sleep -Seconds 10
+              if(((Get-HnsPolicyList).Policies | where InternalPort -EQ 53 | where ExternalPort -EQ 53).Count -ne 2){
+                $policies = ((Get-HnsPolicyList).Policies | where InternalPort -EQ 53 | where ExternalPort -EQ 53)
+                $now = date
+                "$now =========================================" >> $dnsLogs
+                $policies >> $dnsLogs

```

Comment:
We can initiate log collection here, in case we want to reduce the number of manual steps.

Reply:
if the dns failure is due to some other reason, then the control won't come here.